test/end-to-end/test_launcher_restarts_supervisor.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+set -eou pipefail
+
+wait_for_sup_to_start() {
+	until pgrep hab-sup &>/dev/null; do
+		echo -n .
+		sleep 1
+	done
+	echo
+}
+
+if pgrep hab-launch &>/dev/null; then
+	echo "Error: launcher process is already running"
+	exit 1
+fi
+
+TESTING_FS_ROOT=$(mktemp -d)
+export TESTING_FS_ROOT
+sup_log="$TESTING_FS_ROOT/hab/sup/default/sup.log"
+exit_file=$(mktemp)
+exit_code=${1:-1}
+
+mkdir -p "$(dirname "$sup_log")"
+echo -n "Starting launcher with root $TESTING_FS_ROOT (logging to $sup_log)..."
+env HAB_FEAT_TEST_EXIT="$exit_file" hab sup run &> "$sup_log" &
```

Comment:
This will be using whatever hab, supervisor, and launcher packages happen to be installed on the system. Can we take advantage of the `HAB_LAUNCH_BINARY` and `HAB_SUP_BINARY` environment variables to control what gets run?

Reply:
The idea is to pass those environment variables into she test script:
```
➤ env HAB_SUP_BINARY=(pwd)/target/debug/hab-sup HAB_LAUNCH_BINARY=(pwd)/target/debug/hab-launch ./test/end-to-end/test_launcher_restarts_supervisor.sh 
```
That makes it easy to go back and forth between the behavior of the installed system and the local changes.

Reply:
Fair point; can we document that intention in the script?